src/util/spawn_trigger.rs

```diff
@@ -0,0 +1,21 @@
+use bevy::{ecs::system::EntityCommands, prelude::*};
+
+use super::spawn::Spawn;
+
+/// Trait for spawning empty entitie and triggering them.
+pub(crate) trait SpawnTrigger {
+    /// Spawns an empty entity and triggers it with the event.
+    fn spawn_trigger<E: Event>(&mut self, event: E) -> EntityCommands;
+}
+
+impl<T: Spawn> SpawnTrigger for T {
+    fn spawn_trigger<E: Event>(&mut self, event: E) -> EntityCommands {
```

Comment:
Maybe this would read better renamed to `TriggerSpawn::trigger_spawn`?

Reply:
But it spawns then triggers 😛

Reply:
Yeah. Implementation detail though, can read it as "triggering a spawn", which is imo conceptually valid.